### PR TITLE
Create entry on Tensor operation for Pytorch

### LINKED ISSUE #4934: [Concept Entry] PyTorch Tensor Operations
### Reason/inspiration (optional)

We would like a new entry on the `Tensor Operations` concept under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/tensor-operations/tensor-operations.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept (if applicable)
- An `Example` section that provides an example demonstrating the concept in use (if applicable)

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by arisdelacruz**
### Description

This is an attempt to create an entry on the tensor operations for Pytorch in order to help expand the information repository or the subjects on docs.

### Issue Solved

Closes #4934 

### Type of Change

- Adding a new entry

### Checklist

- [ /] All writings are my own.
- [/ ] My entry follows the Codecademy Docs style guide.
- [/ ] My changes generate no new warnings.
- [/ ] I have performed a self-review of my own writing and code.
- [ /] I have checked my entry and corrected any misspellings.
- [/ ] I have made corresponding changes to the documentation if needed.
- [ /] I have confirmed my changes are not being pushed from my forked `main` branch.
- [/ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ /] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @arisdelacruz, the entry is supposed to be a Concept Entry for the Tensor Operations in PyTorch. For the terms that you have added like `expand()`, `permute()`, `tolist()`, etc. separate term entries will be created. So could you modify this based on the concept entry template. 
Sharing one example for your reference - https://www.codecademy.com/resources/docs/numpy/math-methods

Let me know once this is done, or if you have further questions 😄 
Thank you!

**Comment by arisdelacruz** [comment]
thank you for the heads up, @mamtawardhani 

actually, took time to review my entry and, yeah, you're right. Had overdone the same, hehe. My bad.

will revise, thanks

**Comment by mamtawardhani** [comment]
> thank you for the heads up, @mamtawardhani
> 
> actually, took time to review my entry and, yeah, you're right. Had overdone the same, hehe. My bad.
> 
> will revise, thanks

No problem 😃

**Comment by arisdelacruz** [comment]
@mamtawardhani 

just tried my luck. Thanks much.

File: content/pytorch/concepts/tensor-operations/tensor-operations.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Tags:
+  - 'Deep Learning'
+  - 'TensorFlow'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'py-torch-for-classification'
+  - 'paths/data-science'
+---
+
+**Pytorch Tensor Operations** are fundamentals for performing various computations on tensors. Tensors are multi-dimensional arrays that can be manipulated using a wide range of operations.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/tensor-operations.md:16]:
```suggestion
In PyTorch, **Tensor Operations** are fundamentals for performing various tensor computations. Tensors are multi-dimensional arrays that can be manipulated using a wide range of operations.
```

File: content/pytorch/concepts/tensor-operations/tensor-operations.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Some Base Operations
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/tensor-operations.md:18]:
```suggestion
## Fundamental Tensor Operations
```

File: content/pytorch/concepts/tensor-operations/tensor-operations.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+The `expand()` operation is used to expand the tensor into a number of tensors, a number of rows in tensors, and a number of columns in tensors.
+
+The `permute()` operation is used to reorder the tensor using row and column.
+
+The `tolist()` method is used to return a list or nested list from the given tensor.
+
+This `narrow()` function is used to narrow the tensor. in other words, it will extend the tensor based on the input dimensions.
+
+The `where()` function is used to return the new tensor by checking the existing tensors conditionally.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/tensor-operations.md:28]:
```suggestion
- `expand()`: Expands the tensor along specified dimensions, creating a larger tensor with repeated values. 
- `permute()`: Reorders the dimensions of the tensor according to a specified order.
- `tolist()`: Converts the tensor to a Python list or nested list.
- `narrow()`: Returns a tensor that is a narrowed view of the original tensor based on specified dimensions.
- `where()`: Returns a new tensor by applying a condition to the existing tensor.
```

File: content/pytorch/concepts/tensor-operations/tensor-operations.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- `*`: Multiplication
+- `/`: Division
+
+## Element-wise Operations
+
+Element-wise operations are operations that are applied to each element of a tensor individually. These operations include `exponentiation`, `square root`, `absolute value`, among others.
+
+## Reduction Operations
+
+Reduction operations are operations that reduce the number of elements in a tensor by performing a computation across the tensor. These operations include `sum`, `mean`, `maximum`, `minimum`, among others.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/tensor-operations.md:45]:
```suggestion
Reduction operations compute a single result from multiple tensor elements. These operations include:
- `sum()`: Calculates the sum of all elements.
- `mean()`: Computes the mean of elements.
- `max()`: Finds the maximum value.
- `min()`: Finds the minimum value.
```

File: content/pytorch/concepts/tensor-operations/tensor-operations.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+Reshaping in tensor operations is the process of changing the shape or dimensions of a tensor while preserving its underlying data. It involves rearranging the elements of a tensor to fit a new shape, without changing the total number of elements.
+
+We can perform Reshaping using the `torch.reshape()` function or the `.view()` method.
+
+### Concatenation
+
+Concatenation in tensor operations is the process of joining two or more tensors along a specific dimension to form a larger tensor. The resulting tensor has a new dimension that is the concatenation of the original dimensions of the input tensors.
+
+It can be performed using the `torch.cat()` function.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/tensor-operations.md:71]:
```suggestion
Advanced tensor operations include the following:
- Matrix Multiplication:  Performed using `torch.mm()` or the `@` operator.
- Transposition: Flips the dimensions of a tensor. For 2D tensors, it exchanges rows and columns. Achieved using `torch.t()`.
- Reshaping: Changes the shape of a tensor while preserving its data. This can be done using `torch.reshape()` or the `.view()` method.
- Concatenation Joins two or more tensors along a specified dimension. This can be performed using `torch.cat()`.
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @arisdelacruz thank you for your contribution to Codecademy Docs, the entry is nicely written 😄 

I've suggested a few modifications (mostly to convert them to bullet points), could you please review and modify them at your earliest convenience? Thank you! 😃

**Comment by arisdelacruz** [comment]
@mamtawardhani thanks much

**Review by mamtawardhani — APPROVED** [review]
Thank you for your contribution @arisdelacruz 😃 

The entry looks good for a second round of review! 🚀
